### ISSUE TITLE
HelloWorld sample source maps support

### ISSUE THREAD

**Issue by axls** [google/j2cl#10]
Does the current version of j2cl support sourcemaps?

Chrome's DevTool doesn't show HelloWorld.java in sources list. Also I see next log output in terminal when launching dev server through ibazel:
```
INFO: 750µs /0:0:0:0:0:0:0:1:38796 404 GET /HelloWorld.js.map
```
It seems like reference to the sources presents in js, but the web server cannot find requested resource.

**Comment by jDramaix** [comment]
Yes, J2CL has sourcemaps support. The web server used in the example is serving correctly the file. We will fix that.

**Comment by gkdn** [comment]
As an update; we are working switching from zip files to TreeArtifacts so after that this should be easier to address.

**Comment by niloc132** [comment]
Worth pointing out perhaps that closure-compiler's BUNDLE mode presently doesn't handle sourcemaps at all, see https://github.com/google/closure-compiler/issues/3247. In that issue, I link the patch we're using in our closure-compiler fork so that j2cl-maven-plugin can provide sourcemaps in dev mode. This is not a perfect solution, but probably will suffice until BUNDLE gets to be more efficient or is replaced in the maven plugin.
